# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../linux/surface-dock-eth.patch
# Copyright (C) 2014 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

The Microsoft Surface Docks uses a R8152, too.

  - Rene Rebe <rene@exactcode.de>

--- linux-3.15/drivers/net/usb/cdc_ether.c.orig	2014-06-16 18:01:08.153110776 +0200
+++ linux-3.15/drivers/net/usb/cdc_ether.c	2014-06-16 18:03:21.465111094 +0200
@@ -659,6 +659,20 @@
 	.driver_info = 0,
 },
 
+/* Microsoft Surface 2 Dock Ethernet Adapter */
+{
+	USB_DEVICE_AND_INTERFACE_INFO(0x045e, 0x07ab, USB_CLASS_COMM,
+			USB_CDC_SUBCLASS_ETHERNET, USB_CDC_PROTO_NONE),
+	.driver_info = 0,
+},
+
+/* Microsoft Surface 3 Dock Ethernet Adapter */
+{
+       USB_DEVICE_AND_INTERFACE_INFO(0x045e, 0x07c6, USB_CLASS_COMM,
+                       USB_CDC_SUBCLASS_ETHERNET, USB_CDC_PROTO_NONE),
+       .driver_info = 0,
+},
+
 /* Samsung USB Ethernet Adapters */
 {
 	USB_DEVICE_AND_INTERFACE_INFO(SAMSUNG_VENDOR_ID, 0xa101, USB_CLASS_COMM,
--- linux-3.15/drivers/net/usb/r8152.c.orig	2014-06-16 18:00:54.873110744 +0200
+++ linux-3.15/drivers/net/usb/r8152.c	2014-06-16 18:04:45.721111295 +0200
@@ -459,6 +459,10 @@
 #define PRODUCT_ID_RTL8152		0x8152
 #define PRODUCT_ID_RTL8153		0x8153
 
+#define VENDOR_ID_MICROSOFT		0x045e
+#define PRODUCT_ID_MICROSOFT2		0x07ab
+#define PRODUCT_ID_MICROSOFT3		0x07c6
+
 #define VENDOR_ID_SAMSUNG		0x04e8
 #define PRODUCT_ID_SAMSUNG		0xa101
 
@@ -3381,6 +3384,23 @@
 		}
 		break;
 
+	case VENDOR_ID_MICROSOFT:
+		switch (id->idProduct) {
+		case PRODUCT_ID_MICROSOFT2:
+		case PRODUCT_ID_MICROSOFT3:
+			ops->init		= r8152b_init;
+			ops->enable		= rtl8152_enable;
+			ops->disable		= rtl8152_disable;
+			ops->up			= r8152b_exit_oob;
+			ops->down		= rtl8152_down;
+			ops->unload		= rtl8152_unload;
+			ret = 0;
+			break;
+		default:
+			break;
+		}
+		break;
+
 	case VENDOR_ID_SAMSUNG:
 		switch (id->idProduct) {
 		case PRODUCT_ID_SAMSUNG:
@@ -3512,6 +3531,8 @@
 static struct usb_device_id rtl8152_table[] = {
 	{USB_DEVICE(VENDOR_ID_REALTEK, PRODUCT_ID_RTL8152)},
 	{USB_DEVICE(VENDOR_ID_REALTEK, PRODUCT_ID_RTL8153)},
+	{USB_DEVICE(VENDOR_ID_MICROSOFT, PRODUCT_ID_MICROSOFT2)},
+	{USB_DEVICE(VENDOR_ID_MICROSOFT, PRODUCT_ID_MICROSOFT3)},
 	{USB_DEVICE(VENDOR_ID_SAMSUNG, PRODUCT_ID_SAMSUNG)},
 	{}
 };
